src/basho_bench_duration.erl

```diff
@@ -96,20 +96,32 @@ terminate(Reason, #state{duration=DurationMins}) ->
         WSup ->
             WRef = erlang:monitor(process, WSup),
             supervisor:terminate_child(basho_bench_run_sup, basho_bench_worker_sup),
+            supervisor:terminate_child(basho_bench_run_sup, basho_bench_stats),
             receive
                 {'DOWN', WRef, process, _Object, _Info} ->
                     ok
             end
     end,
     run_hook(basho_bench_config:get(post_hook, no_op)),
+    %% If eprof_file is not empty, run eprofiling and write output to filename
+    case EprofFile = basho_bench_config:get(eprof_file, "") of 
+        "" ->
+            ok;
+        _ ->
+            io:format(standard_error,"Stopping eprof profiling",[]),
+            eprof:stop_profiling(),
+            eprof:log(EprofFile),
+            eprof:analyze(total),
+            io:format(standard_error,"Eprof output in ~p\n",[EprofFile])
+    end,
     supervisor:terminate_child(basho_bench_sup, basho_bench_run_sup),
+    application:set_env(basho_bench_app, is_running, false),
```

Comment:
From printfs I was finding that it never gets past terminate the basho_bench_run_sup in any event. That may be an artifact of my current runs. Originally the eprof-stop/output code was in basho_bench_app:stop but I found it wasn't making it that far even when I included call to stop things. I can poke at that issue a little more.